executor/aggfuncs/func_avg.go

```diff
@@ -280,6 +288,62 @@ func (e *avgOriginal4DistinctDecimal) AppendFinalResult2Chunk(sctx sessionctx.Co
 	return nil
 }
 
+type avgPartial4DistinctDecimal struct {
+	avgOriginal4DistinctDecimal
+}
+
+func (e *avgPartial4DistinctDecimal) UpdatePartialResult(sctx sessionctx.Context, rowsInGroup []chunk.Row, pr PartialResult) (memDelta int64, err error) {
+	return 0, terror.ClassOptimizer.New(mysql.ErrInternal, mysql.MySQLErrName[mysql.ErrInternal]).GenWithStack("avgPartial4DistinctDecimal.UpdatePartialResult should not be called")
+}
+
+func (e *avgPartial4DistinctDecimal) MergePartialResult(sctx sessionctx.Context, src, dst PartialResult) (memDelta int64, err error) {
+	p1, p2 := (*partialResult4AvgDistinctDecimal)(src), (*partialResult4AvgDistinctDecimal)(dst)
+	if p1.count == 0 {
+		return 0, nil
+	}
+	for i := range p1.valList {
```

Comment:
`MergePartialResult` is single-thread, this heavy calculation should be avoided.